cmd/satellite/main.go

```diff
@@ -127,6 +145,88 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 		filepath.Join(setupCfg.BasePath, "config.yaml"), o)
 }
 
+func cmdDiag(cmd *cobra.Command, args []string) (err error) {
+	// open the psql db
+	dbpath := diagCfg.BasePath
+	s, err := bwagreement.NewServer("postgres", dbpath, zap.NewNop())
+	if err != nil {
+		fmt.Println("Storagenode database couldnt open:", dbpath)
+		return err
+	}
+	//get all bandwidth aggrements entries already ordered
+	rows, err := s.GetBandwidthAllocations(context.Background())
+	if err != nil {
+		fmt.Println("reading bandwidth table error")
+		return err
+	}
+
+	const padding = 3
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, padding, ' ', tabwriter.AlignRight|tabwriter.Debug)
+	fmt.Fprintln(w, "UplinkID\tTotal\t# Of Transactions\tPUT Action\tGET Action\t")
+
+	// Agreement is a struct that contains a bandwidth agreement and the associated signature
+	type UplinkAttributes struct {
+		TotalBytes        int64
+		PutActionCount    int64
+		GetActionCount    int64
+		TotalTransactions int64
+		// additional attributes add here ...
+	}
+
+	uplinkID := make(map[string]UplinkAttributes)
+	uplinkAtt := UplinkAttributes{}
+	var currUplinkID, lastUplinkID string
+
+	for _, row := range rows {
+		// deserializing rbad you get payerbwallocation, total & storage node id
+		rbad := &pb.RenterBandwidthAllocation_Data{}
+		if err := proto.Unmarshal(row.Data, rbad); err != nil {
+			return err
+		}
+		total := rbad.GetTotal()
+
+		// deserializing pbad you get satelliteID, uplinkID, max size, exp, serial# & action
+		pbad := &pb.PayerBandwidthAllocation_Data{}
+		if err := proto.Unmarshal(rbad.GetPayerAllocation().GetData(), pbad); err != nil {
+			return err
+		}
+		currUplinkID = fmt.Sprintf("%s", pbad.GetUplinkId())
+		action := pbad.GetAction()
+
+		if strings.Compare(currUplinkID, lastUplinkID) != 0 {
+			// make an entry
+			uplinkAtt.TotalBytes = total
+			if action == pb.PayerBandwidthAllocation_PUT {
+				uplinkAtt.PutActionCount++
+			} else {
+				uplinkAtt.GetActionCount++
+				fmt.Println(uplinkAtt.GetActionCount)
```

Comment:
Unnecessary print.

Reply:
Made changes similar to, as suggested in storagenode diag tool